### PR TITLE
h2o predictor: adding a Java component (for mlcomp) which uses an h2o Model

### PR DESCRIPTION
**Pull request by liororama**
Adding the h2o_predictor component (Java) for generating batch predictions given input file and h2o Mojo model

### CODE REVIEW

File: components/Java/h2o_predictor/component.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      "key": "convert_unknown_categorical_levels_to_na",
+      "label": "Convert Unknown Categorical Levels To Na",
+      "description": "Convert Unknown Categorical Levels To Na",
+      "type": "boolean",
+      "default": 1,
+      "optional": true
+    },
+    {
+      "key": "convert_invalid_numbers_to_na",
+      "label": "Convert Invalid Numbers To Na",
```
Comment (PMjoydas) [inline components/Java/h2o_predictor/component.json:65]: Na => NaN ?

File: components/Java/h2o_predictor/pom.xml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        <componentDirectory>${project.build.directory}/${project.artifactId}</componentDirectory>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>3.8.1</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (PMjoydas) [inline components/Java/h2o_predictor/pom.xml]: not sure we need the "junit"

File: components/Java/h2o_predictor/prostate-med-test.csv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+AGE,RACE,DCAPS,VOL,GLEASON
+68,2,2,0,6
+71,1,2,0,8
```
Comment (PMjoydas) [inline components/Java/h2o_predictor/prostate-med-test.csv:3]: Please remove this file

File: components/Java/h2o_predictor/src/main/java/com/parallelm/components/h2o_predictor/H2OPredictor.java [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+   private static Logger logger = Logger.getLogger(H2OPredictor.class);
+
+   private final String tmpDir = "/tmp";
+
+    private void checkArgs(List<Object> parentDataObjects) throws Exception {
+
+        String inputSamplesFileStr;
+        // From params
+        String modelPathStr = (String) params.get("input_model");
+        System.out.println("param - input_model:    " + modelPathStr);
```
Comment (PMjoydas) [inline components/Java/h2o_predictor/src/main/java/com/parallelm/components/h2o_predictor/H2OPredictor.java:150]: all System.out.prints, need to be replaced with "logger"

File: components/Java/h2o_predictor/src/test/main/java/com/parallelm/components/h2o_predictor/H2OPredictorTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                out.println(String.format("Expected file (samples:%s) not found", e.getCause()));
+            } else {
+                Assert.fail("Fail");
+            }
+        }
+    }
+
+    @Test
+    public void testH2OmodelImport() throws Exception{
+        out.println("Test 1: testH2OmodelImport in progress.. ");
```
Comment (yakov-g) [inline components/Java/h2o_predictor/src/test/main/java/com/parallelm/components/h2o_predictor/H2OPredictorTest.java]: should it be "Test 3" in ?

File: components/Java/h2o_predictor/src/main/java/com/parallelm/components/h2o_predictor/H2OPredictor.java [hunk 6]
```diff
@@ -194,13 +193,16 @@ private void checkArgs(List<Object> parentDataObjects) throws Exception {
 
     }
 
-    private void loadModel() throws Exception {
+    //private void loadModel() throws Exception {
```
Comment (yakov-g) [inline components/Java/h2o_predictor/src/main/java/com/parallelm/components/h2o_predictor/H2OPredictor.java]: remove line

File: components/Java/h2o_predictor/src/main/java/com/parallelm/components/h2o_predictor/H2OPredictor.java [hunk 7]
```diff
@@ -312,6 +314,8 @@ public static void main(String[] args ) throws Exception {
         }
         System.out.println(options);
 
+        out.println("options:  "+ options);
```
Comment (yakov-g) [inline components/Java/h2o_predictor/src/main/java/com/parallelm/components/h2o_predictor/H2OPredictor.java:316]: there are two spaces after "options", remove one please :)

File: components/Java/h2o_predictor/src/test/main/java/com/parallelm/components/h2o_predictor/H2OPredictorTest.java (hunk 5 above)
Comment (PMjoydas) [reply components/Java/h2o_predictor/src/test/main/java/com/parallelm/components/h2o_predictor/H2OPredictorTest.java]: fixed it, in the following patch

File: components/Java/h2o_predictor/src/test/main/java/com/parallelm/components/h2o_predictor/H2OPredictorTest.java [hunk 8]
```diff
@@ -174,6 +174,40 @@ public void testH2OPredict() throws Exception{
...
         }
     }
-}
 
+    @Test
+    public void testH2OPredictResults() throws Exception{
+        String output_file = "/tmp/out-mini.csv";
+
+        try {
+            out.println("Test 7: testH2OPredictResults in progress.. ");
```
Comment (yakov-g) [inline components/Java/h2o_predictor/src/test/main/java/com/parallelm/components/h2o_predictor/H2OPredictorTest.java]: To tell the truth: why do you want to print "Test NN" is running.. bla bla?
- Test framework should print it by itself.
- And so why to print "Test passed"? If there are no errors, means it passed. And if not - it failed.
- You hardcoded func names into several messages. You could use 
smth like this instead 
https://stackoverflow.com/questions/473401/get-name-of-currently-executing-test-in-junit-4
maybe something like this also available for the junit you use...

**Review by yakov-g — APPROVED** [review]
Approved, but decide if you want to fix printlines. Usually tests evolve and printlns stay the same :)

File: components/Java/h2o_predictor/src/test/main/java/com/parallelm/components/h2o_predictor/H2OPredictorTest.java (hunk 8 above)
Comment (PMjoydas) [reply components/Java/h2o_predictor/src/test/main/java/com/parallelm/components/h2o_predictor/H2OPredictorTest.java]: @yakov-g that's an excellent suggestion, let me work on making these changes

**Pull request merged**